### ISSUE TITLE
Uploading music with the MusicManager gets an error most of the time

### ISSUE THREAD

**Issue by jimwd** [simon-weber/gmusicapi#549]
Hi,

For the past two weeks, I've gotten errors on songs that aren't matched in any manner (or full match or just tag suggestions). When uploading these songs i get a wall of text filling the whole terminal and in the end it says:

<a href=//www.google.com/><span id=logo aria-label=Google></span></a>\                                                           n  <p><b>502.</b> <ins>That\u2019s an error.</ins>\n  <p>The server encountered                                                            a temporary error and could not complete your request.<p>Please try again in 30                                                            seconds.  <ins>That\u2019s all we know.</ins>\n')

When trying again 30seconds later it gives me another error and after with a lot of luck the upload finishes.
It might actually not be the problem of gmusicapi but from Google itself, but normal uploading through the browser works fine..

Jim

**Comment by simon-weber** [comment]
Yeah, I think this is just flakiness on Google's end, especially if retrying the same track works later. I'll leave this open in case other folks notice it too, but I haven't noticed anything through CI or irc lately.

**Comment by jimwd** [comment]
Hi,

Yeah well, now it is not uploading these songs anymore.. I think it is just Google acting up as the MusicManager is now uploading these songs with completely different tags while when it finally goes through with gmusicapi these tags are correctly. The upload through the browser just gets stuck anyways. Maybe add an option to repeat until done (with 1min in between or so) if this 502 is received?

Also for some songs that used a different tagger, it gets a cannot parse date md for xxx (date must be in 1..12). Any recommendations on how to solve that? Read on google that i had to update that parser but it seems to be up to date.

Jim

**Comment by simon-weber** [comment]
Huh, interesting. Yeah, retry logic on our end may be a reasonable bet.

For the other error, can you open a new issue with the full traceback and example input? I haven't heard of it before.

**Comment by jimwd** [comment]
This issue is back after being fine for a few weeks!
